### ISSUE TITLE
How should we implement the feature to control spaces betwteen han/kana and alphanumerics in comments (especially for documentation e.g. JSDoc and DocString), string literals, and HTML?

### ISSUE THREAD

**Issue by tats-u** [prettier/prettier#13145]
Those who wants to add such spaces in Markdown will also want to add ones in strings and comments in other languages.
I have created a plugin to prevent those spaces from inserted in Markdown, but I had to reuse Prettier's entire source code as a submodule and feel that it is difficult to reuse the process that controls those spaces in other languages.
I wish we could hook something in third party plugins to fix this issue because Prettier WILL NEVER add new options ANYMORE, but Prettier seems not to be designed to do so.

**Comment by xsrvmy** [comment]
I don't think this even belongs in prettier. Changing the spacing between character affects the output of code.
